src/components/Carousel/Settings.ts

```diff
@@ -0,0 +1,14 @@
+type IntersectionObserverEntryVisibilityRatio = number;
+type throttleInterval = number;
+
+interface IntersectionObserverOptions extends IntersectionObserverInit {
+    ratio: IntersectionObserverEntryVisibilityRatio;
```

Comment:
IntersectionObserverEntryVisibilityRatio
this means what? The amount of the item visible to be considered visible?

Reply:
Yes, see https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserverEntry/intersectionRatio and how it's used in `src/components/Carousel/Tests/autoScrollApi.test.ts`, formerly `createApi.test.ts`,